src/properties.js

```diff
@@ -458,6 +458,55 @@ export function suspendableProp(options) {
     )
 }
 
+export function delayableHandler(options) {
+  /*
+  Delays `[handler]` calls until after `[filter]` become true.
+  */
+  const handler = options.handler
+  const filter = options.filter
+  return (Component) =>
+    setWrapperName(
+      Component,
+      class delayableHandler extends BaseComponent {
+        constructor(props) {
+          super(props)
+          this.state = {
+            shouldTrigger: false,
+            value: props.value,
+          }
+          this.trigger = () => {
+            const {
+              props: { [handler]: triggerFunction, [filter]: value },
+            } = this
+            if (value) {
+              triggerFunction()
+              this.setState({ shouldTrigger: false })
+            } else {
+              this.setState({ shouldTrigger: true })
+            }
+          }
+        }
+        componentDidUpdate() {
+          const {
+            props: { [handler]: triggerFunction, [filter]: newValue },
+            state: { shouldTrigger, value },
+          } = this
+          if (shouldTrigger && newValue && newValue !== value) {
+            triggerFunction()
+            this.setState({ value: newValue })
```

Comment:
No need to track `sentinel` value changes. However, `shouldTrigger` should be set back to `false`, and `handler` (triggerFunction) should be called after the state is updated:
```suggestion
            this.setState({ shouldTrigger: false }, handler)
```